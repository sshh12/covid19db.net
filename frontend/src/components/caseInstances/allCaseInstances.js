import CaseInstanceGBR from './caseInstance0';
import CaseInstanceMEX from './caseInstance1';
import CaseInstanceUSA from './caseInstance2';

export {
    CaseInstanceGBR,
    CaseInstanceMEX,
    CaseInstanceUSA
}