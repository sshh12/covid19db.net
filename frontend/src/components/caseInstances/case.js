class Case {

  constructor(code, name, date, location, newCases, percentages, derivatives, history) {
    this.code = code;
    this.name = name;
    this.date = date;
    this.location = location;
    this.newCases = newCases;
    this.percentages = percentages;
    this.derivatives = derivatives;
    this.history = history;
  }

  static fromJSON(json) {
    var obj = JSON.parse(json);
    return new Case(
      obj.country.codes.alpha3Code,
      obj.country.name,
      obj.date,
      obj.location,
      obj.totals,
      obj.new,
      obj.percentages,
      obj.derivativeNew,
      obj.history
    );
  }

}
