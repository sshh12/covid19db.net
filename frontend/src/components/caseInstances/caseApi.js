import axios from 'axios';

const url = 'http://localhost:5000/case-statistics';

class CaseApi {

  constructor() {
  }

  getAll(attributes) {
    var getURL = url + "?attributes=";
    for (var i = 0; i < attributes.length; i++) {
      getURL += attributes[i];
      getURL += ",";
    }
    getURL = getURL.substring(0, getURL.length - 1);

    var response = fetch(getURL).then(response => response.json());
    return response;
  }

  getInstance(country, attributes) {
    var getURL = url + `/${country}`;
    if (attributes.length > 0) {
      getURL += '?attributes=';
      for (var i = 0; i < attributes.length; i++) {
        getURL += attributes[i];
        getURL += ",";
      }
      getURL = getURL.substring(0, getURL.length - 1);
    }
    console.log(getURL);
    // var response = fetch(getURL).then(response => response.json());
    axios.get(getURL)
      .then(res => {
        console.log(res.data);
      })
    //return response;
  }

}

export default CaseApi;
