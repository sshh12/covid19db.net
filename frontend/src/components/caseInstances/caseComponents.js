import React, { Component } from 'react';
import { ResponsiveLine } from '@nivo/line';
import "./caseInstance.css";

function Totals(props) {
  return <div className='totals'>
    <h2 className='totals-title'>{props.title}</h2>
    <h2 className='totals-data'>{props.data}</h2>
  </div>
}

function NewStats(props) {
  return <div className='new-stats'>
    <h2 className='new-stats-title'>{props.title}</h2>
    <h2 className='new-stats-data'>{props.data}</h2>
    <h2 className='new-stats-title'>{props.yesterday} from yesterday</h2>
  </div>
}

function GenStats(props) {
  return <div className='new-stats'>
    <h2 className='new-stats-title'>{props.title}</h2>
    <h2 className='new-stats-data'>{props.data}%</h2>
    <h2 className='gen-stats-description'>{props.description}</h2>
  </div>
}

function compileData(data) {
  var history = data.history;
  var confirmed = [];
  var deaths = [];
  var recovered = [];
  var active = [];
  var dailyCases = [];
  for (var i = 7; i < history.length; i += 3) {
    var date = i;//history[i].Date;
    //date = 0
    dailyCases.push({ "x": date, "y": history[i].Confirmed - history[i - 7].Confirmed });
    confirmed.push({ "x": date, "y": history[i].Confirmed });
    deaths.push({ "x": date, "y": history[i].Deaths });
    recovered.push({ "x": date, "y": history[i].Recovered });
    active.push({ "x": date, "y": history[i].Active });

  }

  var lineData = [];
  lineData.push({ "id": 'New Cases', "data": dailyCases });
  // lineData.push({ "id": 'Confirmed', "data": confirmed });
  // lineData.push({ "id": 'Deaths', "data": deaths });
  // lineData.push({ "id": 'Recovered', "data": recovered });
  // lineData.push({ "id": 'Active', "data": active });

  console.log(lineData);
  return lineData;
}

function CaseResponseLine(props) {
  return <ResponsiveLine
    data={compileData(props.data)}
    margin={{ top: 50, right: 110, bottom: 50, left: 60 }}
    xScale={{ type: 'point' }}
    yScale={{ type: 'linear', min: 'auto', max: 'auto', stacked: true, reverse: false }}
    axisTop={null}
    axisRight={null}
    axisBottom={{
      orient: 'bottom',
      tickSize: 5,
      tickPadding: 5,
      tickRotation: 0,
      legend: 'Date',
      legendOffset: 36,
      legendPosition: 'middle'
    }}
    axisLeft={{
      orient: 'left',
      tickSize: 5,
      tickPadding: 5,
      tickRotation: 0,
      legend: 'Amount',
      legendOffset: -40,
      legendPosition: 'middle'
    }}
    colors={{ scheme: 'nivo' }}
    pointSize={10}
    pointColor={{ theme: 'background' }}
    pointBorderWidth={2}
    pointBorderColor={{ from: 'serieColor' }}
    pointLabel="y"
    pointLabelYOffset={-12}
    useMesh={true}
    legends={[
      {
        anchor: 'bottom-right',
        direction: 'column',
        justify: false,
        translateX: 100,
        translateY: 0,
        itemsSpacing: 0,
        itemDirection: 'left-to-right',
        itemWidth: 80,
        itemHeight: 20,
        itemOpacity: 0.75,
        symbolSize: 12,
        symbolShape: 'circle',
        symbolBorderColor: 'rgba(0, 0, 0, .5)',
        effects: [
          {
            on: 'hover',
            style: {
              itemBackground: 'rgba(0, 0, 0, .03)',
              itemOpacity: 1
            }
          }
        ]
      }
    ]}
  />
}


export {
  Totals,
  NewStats,
  GenStats,
  CaseResponseLine
}