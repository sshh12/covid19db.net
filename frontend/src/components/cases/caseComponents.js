import CaseResponseLine from "./components/caseResponseLine";
import GenStats from "./components/genStats";
import NewStats from "./components/newStats";
import Totals from "./components/totals";
import CaseButtonGroup from "./components/caseButtonGroup";
import CaseComparison from "./components/caseComparison";
import CaseComparisonCollapse from "./components/caseComparisonCollapse";
import SearchBar from "../search/casesSearchBar";

export {
  CaseResponseLine,
  GenStats,
  NewStats,
  Totals,
  CaseButtonGroup,
  CaseComparison,
  CaseComparisonCollapse,
  SearchBar,
};
