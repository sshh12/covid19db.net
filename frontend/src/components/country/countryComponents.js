import React from 'react';
import { Card, Col } from 'antd'
import "./countryInstance.css";

const { Meta } = Card;

function GeneralInfo(props) {
  return <div className='new-stats'>
    <h2 className='title' style={{ color: 'grey', fontSize: 'medium' }}>{props.title}</h2>
    <h2 className='data'>{props.data}</h2>
  </div>
}

function News(props) {
  return <div className='news'>
    <a className='title' href={props.url}>{props.title}</a>
    <h2 className='source'>({props.source})</h2>
  </div>
}

function AllNews(news) {
  return news.map(n => 
    <Col>
      <a href={n.url}>
        <Card
          hoverable
          style={{ width: 240 }}
          cover={<img src={n.urlToImage} alt="Article image" />}
          title={n.title}
        >
          <Meta title={n.author} description={n.description} />
        </Card>
      </a>
    </Col>
  )
}

function GetImage(url) {
  var image = document.createElement('img');
  image.src = url;
  document.body.appendChild(image);
}

export {
  GeneralInfo,
  GetImage, News, AllNews
}