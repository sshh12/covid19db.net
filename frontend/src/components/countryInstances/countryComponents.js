import React, { Component } from 'react';
import "./countryInstance.css";

function GeneralInfo(props) {
  return <div className='new-stats'>
    <h2 className='title'>{props.title}</h2>
    <h2 className='data'>{props.data}</h2>
  </div>
}
//TODO add function to display map, flag, etc

export {
  GeneralInfo,
}