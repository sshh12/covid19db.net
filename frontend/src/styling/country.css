.country-page-header {
  background-color: #323776;
  justify-content: center;
  display: flex;
}

.country-page-header-content {
  display: flex;
  flex-direction: column;
  padding-bottom: 40px;
  padding-top: 20px;
  align-items: center;
  justify-content: center;
  width: 75vw;
}
.page-title {
  font-weight: 800;
  margin-bottom: 20px;
  font-size: 2em;
  color: white;
}
.page-description {
  color: white;
  font-size: 1.1em;
  margin-bottom: 20px;
}

.compare-title {
  font-weight: 800;
  font-size: 1.4em;
  color: white;
  text-align: left;
}

.compare-group {
  align-self: start;
  display: flex;
  flex-direction: column;
  justify-content: start;
}
