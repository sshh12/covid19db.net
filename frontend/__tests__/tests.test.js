import React from 'react'
import { shallow } from 'enzyme'
// import App from '../src/app'
// import CountryInstance from '../src/components/country/countryInstance'
// import RiskInstance from '../src/components/risks/riskInstance'
// import CaseInstance from '../src/components/caseInstances/caseInstance'
// import Main from '../src/pages/main'
import About from '../src/pages/about'
import Risk from '../src/pages/risks'
import Countries from '../src/pages/countries'
import Cases from '../src/pages/cases'
// import { Totals, NewStats, GenStats, CaseResponseLine } from '../src/components/caseInstances/caseComponents'
import { Totals, NewStats, GenStats } from '../src/components/caseInstances/caseComponents'
import GlobalNews from '../src/pages/globalNews'
// import CountryCard from '../src/components/country/countryCard'
// import { GeneralInfo, GetImage, News, AllNews } from '../src/components/country/countryComponents'
import { GeneralInfo, GetImage, News } from '../src/components/country/countryComponents'

// Run 'yarn test'
const dummyData = {}
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve(dummyData),
  })
);

// DUMB TESTS
describe('Sample Tests', () => {
  test('Sample Test', () => {
    expect(true).toBeTruthy();
  });

  test('Sample Test 2', () => {
    const x = true;
    expect(x).toBe(true);
  })

});

describe('Render Pages', () => {
  test('About', () => {
    // Render the Demoinstance component
    const test = shallow(<About/>); 
    expect(test).toMatchSnapshot();
  });

  test('Risk', () => {
      // Render the Demoinstance component
      const test = shallow(<Risk/>); 
      expect(test).toMatchSnapshot();
  });
  test('Country', () => {
    // Render the Demoinstance component
    const test = shallow(<Countries/>); 
    expect(test).toMatchSnapshot();
  });
  test('Cases', () => {
    // Render the Demoinstance component
    const test = shallow(<Cases/>); 
    expect(test).toMatchSnapshot();
  });
  // test('Main', () => {
  //   // Render the Demoinstance component
  //   const test = shallow(<Main/>); 
  //   expect(test).toMatchSnapshot();
  // });
  test('Global News', () => {
    // Render the Demoinstance component
    const test = shallow(<GlobalNews/>); 
    expect(test).toMatchSnapshot();
  });

});

describe('Render Components', () => {
  // Country Components

  // TODO commented out bc fails test
  // test('Country: Country Card', () => {
  //   // Render the Demoinstance component
  //   const test = shallow(<CountryCard/>); 
  //   expect(test).toMatchSnapshot();
  // });
  test('Country: General Info', () => {
    // Render the Demoinstance component
    const test = shallow(<GeneralInfo/>); 
    expect(test).toMatchSnapshot();
  });
  test('Country: Get Image', () => {
    // Render the Demoinstance component
    const test = shallow(<GetImage/>); 
    expect(test).toMatchSnapshot();
  });
  test('Country: News', () => {
    // Render the Demoinstance component
    const test = shallow(<News/>); 
    expect(test).toMatchSnapshot();
  });
  // TODO commented out bc fails test
  // test('Country: All News', () => {
  //   // Render the Demoinstance component
  //   const test = shallow(<AllNews/>); 
  //   expect(test).toMatchSnapshot();
  // });

  // Case components
  test('Case: Totals', () => {
    // Render the Demoinstance component
    const test = shallow(<Totals/>); 
    expect(test).toMatchSnapshot();
  });
  test('Case: New Stats', () => {
    // Render the Demoinstance component
    const test = shallow(<NewStats/>); 
    expect(test).toMatchSnapshot();
  });
  test('Case: Gen Stats', () => {
    // Render the Demoinstance component
    const test = shallow(<GenStats/>); 
    expect(test).toMatchSnapshot();
  });
  // TODO commented out bc fails test
  // test('Case: Case Response Line', () => {
  //   // Render the Demoinstance component
  //   const test = shallow(<CaseResponseLine/>); 
  //   expect(test).toMatchSnapshot();
  // });
})
