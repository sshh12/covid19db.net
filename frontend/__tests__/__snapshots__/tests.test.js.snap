// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Components Case: Case Response Line 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
/>
`;

exports[`Render Components Case: Gen Stats 1`] = `
<div
  className="new-stats"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <h2
    className="new-stats-title"
  />
  <h2
    className="new-stats-data"
  >
    %
  </h2>
</div>
`;

exports[`Render Components Case: New Stats 1`] = `
<div
  className="new-stats"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <h2
    className="new-stats-title"
  />
  <h2
    className="new-stats-data"
  />
  <h2
    className="new-stats-title"
  >
     from yesterday
  </h2>
</div>
`;

exports[`Render Components Case: Totals 1`] = `
<div
  className="totals"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <h2
    className="totals-title"
  />
  <h2
    className="totals-data"
  />
</div>
`;

exports[`Render Components Country: All News 1`] = `null`;

exports[`Render Components Country: General Info 1`] = `
<div
  className="new-stats"
>
  <h2
    className="title"
    style={
      Object {
        "color": "grey",
        "fontSize": "medium",
      }
    }
  />
  <h2
    className="data"
  />
</div>
`;

exports[`Render Components Country: News 1`] = `
<div
  className="news"
>
  <a
    className="title"
  />
  <h2
    className="source"
  >
    (
    )
  </h2>
</div>
`;

exports[`Render Pages About 1`] = `
<div
  className="App"
>
  <h1
    style={
      Object {
        "fontSize": "2em",
        "fontWeight": "800",
        "marginBottom": "20px",
        "marginTop": "20px",
      }
    }
  >
    About
  </h1>
  <div
    style={
      Object {
        "marginTop": "18px",
        "paddingBottom": "18px",
      }
    }
  >
    Our COVID-19 Database site allows users to gather quick statistics on the coronavirus in terms of country, cases, and potential risks. This website is intended for those susceptible to contracting the virus (that means you!), and our aim is to keep users informed about the coronavirus. Integrating disparate data sets allows us to form a broad overview of how countries have handled the COVID-19 pandemic, present some of the unique risks each country's population faces, and possibly indicate the near future trajectory of the pandemic.
  </div>
  <hr />
  <div
    style={
      Object {
        "margin": "18px",
      }
    }
  >
    <h4>
      Project Info
    </h4>
    <br />
    <div
      className="row"
    >
      <div
        className="col-sm-4"
      >
        <div
          className="list-group"
          onKeyDown={[Function]}
          role="tablist"
        >
          <a
            aria-selected={true}
            className="list-group-item list-group-item-action active"
            data-rb-event-key="#tools"
            disabled={false}
            href="#tools"
            onClick={[Function]}
            role="tab"
          >
            Tools
          </a>
          <a
            aria-selected={false}
            className="list-group-item list-group-item-action"
            data-rb-event-key="#data"
            disabled={false}
            href="#data"
            onClick={[Function]}
            role="tab"
            tabIndex={-1}
          >
            Data
          </a>
        </div>
      </div>
      <div
        className="col-sm-8"
      >
        <div
          className="tab-content"
        >
          <div
            aria-hidden={false}
            aria-labelledby={null}
            className="fade tab-pane active show"
            id={null}
            role="tabpanel"
            style={
              Object {
                "textAlign": "left",
              }
            }
          >
            <ul
              className="list-unstyled"
            >
              <li>
                <a
                  href="https://reactjs.org/"
                >
                  <img
                    height="100rem"
                    src="/icons/react.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  React
                </a>
                 
                - 
                For creating interactive UIs in JSX.
              </li>
              <li>
                <a
                  href="https://ant.design/"
                >
                  <img
                    height="100rem"
                    src="/icons/antd.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  Antd
                </a>
                 
                - 
                A React UI library for clean composable components.
              </li>
              <li>
                <a
                  href="https://nivo.rocks/"
                >
                  <img
                    height="100rem"
                    src="/icons/nivo.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  Nivo
                </a>
                 
                - 
                A React UI library for animated and interactive charts.
              </li>
              <li>
                <a
                  href="https://jestjs.io/"
                >
                  <img
                    height="100rem"
                    src="/icons/jest.jpg"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  Jest
                </a>
                 
                - 
                A JavaScript testing framework.
              </li>
              <li>
                <a
                  href="https://github.com/axios/axios"
                >
                  <img
                    height="100rem"
                    src="/icons/axios.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  Axios
                </a>
                 
                - 
                For doing browser-side HTTP requests to our API.
              </li>
              <li>
                <a
                  href="https://flask.palletsprojects.com/en/1.1.x/"
                >
                  <img
                    height="100rem"
                    src="/icons/flask.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  Flask
                </a>
                 
                - 
                For serving our API data in a RESTful way.
              </li>
              <li>
                <a
                  href="https://www.sqlalchemy.org/"
                >
                  <img
                    height="100rem"
                    src="/icons/sqla.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  SQLAlchemy
                </a>
                 
                - 
                A Python ORM for creating and interacting with our SQL models.
              </li>
              <li>
                <a
                  href="https://cloud.google.com"
                >
                  <img
                    height="100rem"
                    src="/icons/gcp.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  GCP
                </a>
                 
                - 
                For hosting our site on the cloud.
              </li>
              <li>
                <a
                  href="https://www.postman.com"
                >
                  <img
                    height="100rem"
                    src="/icons/postman.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  Postman
                </a>
                 
                - 
                For building and testing API's.
              </li>
              <li>
                <a
                  href="https://gitlab.com/explore"
                >
                  <img
                    height="100rem"
                    src="/icons/gitlab.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  GitLab
                </a>
                 
                - 
                DevOps tool for issue-tracking, continuous integration, testing, and deployment.
              </li>
              <li>
                <a
                  href="https://www.selenium.dev/"
                >
                  <img
                    height="100rem"
                    src="/icons/selenium.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  Selenium
                </a>
                 
                - 
                For end-to-end automated GUI testing.
              </li>
            </ul>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby={null}
            className="fade tab-pane"
            id={null}
            role="tabpanel"
            style={
              Object {
                "textAlign": "left",
              }
            }
          >
            For the first phase, we downloaded a few local copies of the data provided from these sources and glued them into the instances using either a script or by copy-paste. The COVID-19 API, RestCountries, and NewsAPI all provided APIs to access their data, but the OWID dataset is only available in file format (
            <code>
              .json
            </code>
             in this case).
            <ul
              className="list-unstyled"
            >
              <li>
                <a
                  href="https://api.covid19api.com"
                >
                  <img
                    height="100rem"
                    src="/icons/server.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  COVID-19 API
                </a>
              </li>
              <li>
                <a
                  href="https://restcountries.eu"
                >
                  <img
                    height="100rem"
                    src="/icons/server.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  REST Countries
                </a>
              </li>
              <li>
                <a
                  href="https://newsapi.org"
                >
                  <img
                    height="100rem"
                    src="/icons/server.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  NewsAPI
                </a>
              </li>
              <li>
                <a
                  href="https://covid.ourworldindata.org/data/owid-covid-data.json"
                >
                  <img
                    height="100rem"
                    src="/icons/server.png"
                    style={
                      Object {
                        "marginRight": "10px",
                      }
                    }
                    width="100rem"
                  />
                  OWID COVID-19 Dataset
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr />
  <div
    style={
      Object {
        "marginTop": "18px",
        "paddingBottom": "18px",
      }
    }
  >
    <h4>
      GitLab Statistics
    </h4>
    <div
      className="spinner-border text-primary"
    />
    <br />
    <a
      href="https://gitlab.com/jrmoulckers/covid19db-net"
    >
      <img
        height="100rem"
        src="/icons/gitlab.png"
        width="100rem"
      />
    </a>
    <br />
    <a
      href="https://documenter.getpostman.com/view/12799044/TVKJxuP4"
    >
      <img
        height="100rem"
        src="/icons/postman.png"
        width="100rem"
      />
    </a>
  </div>
  <hr />
  <div>
    <div
      className="card"
      style={
        Object {
          "display": "inline-block",
          "margin": "10px",
          "width": "22rem",
        }
      }
    >
      <a
        href="https://www.linkedin.com/in/shrivushankar/"
      >
        <img
          className="card-img-top"
          src="/imgs/sshh12.jpg"
        />
      </a>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          <a
            href="https://www.linkedin.com/in/shrivushankar/"
          >
            Shrivu Shankar
          </a>
        </div>
        <p
          className="card-text"
        >
          Junior Computer Science Major from Houston, TX
          <hr />
          <b>
            Role
          </b>
           
          Full Stack Developer
           
          <hr />
          <div
            className="spinner-border text-primary"
          />
        </p>
      </div>
    </div>
    <div
      className="card"
      style={
        Object {
          "display": "inline-block",
          "margin": "10px",
          "width": "22rem",
        }
      }
    >
      <a
        href="https://www.linkedin.com/in/cameron-doggett-186761164/"
      >
        <img
          className="card-img-top"
          src="/imgs/camd99.jpg"
        />
      </a>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          <a
            href="https://www.linkedin.com/in/cameron-doggett-186761164/"
          >
            Cameron Doggett
          </a>
        </div>
        <p
          className="card-text"
        >
          Junior Computer Science Major from Katy, TX
          <hr />
          <b>
            Role
          </b>
           
          Back-end Developer
           
          <hr />
          <div
            className="spinner-border text-primary"
          />
        </p>
      </div>
    </div>
    <div
      className="card"
      style={
        Object {
          "display": "inline-block",
          "margin": "10px",
          "width": "22rem",
        }
      }
    >
      <a
        href="https://www.linkedin.com/in/yashkakodkar/"
      >
        <img
          className="card-img-top"
          src="/imgs/YashKakodkar.jpg"
        />
      </a>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          <a
            href="https://www.linkedin.com/in/yashkakodkar/"
          >
            Yash Kakodkar
          </a>
        </div>
        <p
          className="card-text"
        >
          Junior Computer Science Major from Katy, TX
          <hr />
          <b>
            Role
          </b>
           
          Full Stack Developer
           
          <hr />
          <div
            className="spinner-border text-primary"
          />
        </p>
      </div>
    </div>
    <div
      className="card"
      style={
        Object {
          "display": "inline-block",
          "margin": "10px",
          "width": "22rem",
        }
      }
    >
      <a
        href="https://www.linkedin.com/in/jeffreymoulckers/"
      >
        <img
          className="card-img-top"
          src="/imgs/jrmoulckers.jpg"
        />
      </a>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          <a
            href="https://www.linkedin.com/in/jeffreymoulckers/"
          >
            Jeffrey Moulckers
          </a>
        </div>
        <p
          className="card-text"
        >
          Junior Computer Science/Spanish Dual Major from Austin, TX
          <hr />
          <b>
            Role
          </b>
           
          Full Stack Developer
           
          <b>
            (Team Lead)
          </b>
          <hr />
          <div
            className="spinner-border text-primary"
          />
        </p>
      </div>
    </div>
    <div
      className="card"
      style={
        Object {
          "display": "inline-block",
          "margin": "10px",
          "width": "22rem",
        }
      }
    >
      <a
        href="https://www.linkedin.com/in/vassi-gianitsos-a3881b167/"
      >
        <img
          className="card-img-top"
          src="/imgs/vassi_gianitsos.jpg"
        />
      </a>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          <a
            href="https://www.linkedin.com/in/vassi-gianitsos-a3881b167/"
          >
            Vassi Gianitsos
          </a>
        </div>
        <p
          className="card-text"
        >
          Senior Computer Science Major from Flower Mound, TX
          <hr />
          <b>
            Role
          </b>
           
          Front-end Developer
           
          <hr />
          <div
            className="spinner-border text-primary"
          />
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render Pages Cases 1`] = `
<div
  style={
    Object {
      "height": "50vh",
      "paddingTop": 250,
      "textAlign": "center",
      "width": "100vw",
    }
  }
>
  <div
    className="ant-spin ant-spin-lg ant-spin-spinning"
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
    </span>
  </div>
</div>
`;

exports[`Render Pages Country 1`] = `
<div
  style={
    Object {
      "height": "50vh",
      "paddingTop": 250,
      "textAlign": "center",
      "width": "100vw",
    }
  }
>
  <div
    className="ant-spin ant-spin-lg ant-spin-spinning"
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
    </span>
  </div>
</div>
`;

exports[`Render Pages Global News 1`] = `<div />`;

exports[`Render Pages Main 1`] = `
<div
  className="App"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div
    className="ant-spin ant-spin-lg ant-spin-spinning"
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
    </span>
  </div>
</div>
`;

exports[`Render Pages Risk 1`] = `
<div
  style={
    Object {
      "height": "50vh",
      "paddingTop": 250,
      "textAlign": "center",
      "width": "100vw",
    }
  }
>
  <div
    className="ant-spin ant-spin-lg ant-spin-spinning"
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
    </span>
  </div>
</div>
`;
